stdlib/LinearAlgebra/src/lbt.jl

```diff
@@ -171,9 +171,34 @@ function Base.show(io::IO, mime::MIME{Symbol("text/plain")}, lbt::LBTConfig)
     end
 end
 
+mutable struct ConfigCache
+    # We only store `Union{Nothing,LBTConfig}`.  However, declare the field type
+    # to be `Any` (pointer indirection) for lock-free store and load.
+    @atomic config::Any
```

Comment:
@vtjnash Does this make sense or do we want to drop c6acae5c9b460cc4c7807e612712b92adfaebb61?

Reply:
We use pointer indirection already for this ABI with Union, for the reason in the comment here

Reply:
Ah, that's good to know! I probably should've just looked at the LLVM IR but I thought we were generating a lock when the object size exceeds the hardware limit.